clang/test/C/C23/n3042.c

```diff
@@ -93,6 +91,9 @@ void test() {
   (int *)null_val;    // ok
   (int *)nullptr;     // ok
   (nullptr_t)nullptr; // ok
+  (nullptr_t)0;       // ok
```

Comment:
Maybe worth checking other literals as well, like floating point, char and string literals?

Reply:
I added that test coverage in fb9deab74e5dc9a9227732fcd95c1aadacf86d44, thanks for the suggestion!